qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,335 @@
+        for constraint in problem.linear_constraints:
+
+            # special constraint check function here
+            if not self._is_special_constraint(problem, constraint):
+                self._dst.linear_constraints.append(constraint)
+                continue
+            #
+
+            conv_matrix = self._conversion_matrix(constraint)
+            rowlist = list(constraint.linear.to_dict().items())
+
+            # constant part
+            if conv_matrix[0][0] != 0:
+                offset += sense * penalty * conv_matrix[0][0]
+
+            # linear parts of penalty
+            for (j, z) in enumerate(rowlist):
+                # if j already exists in the linear terms dic, add a penalty term
+                # into existing value else create new key and value in the linear_term dict
+
+                if conv_matrix[0][j + 1] != 0:
+                    linear[z[0]] = linear.get(z[0], 0.0) + sense * penalty * conv_matrix[0][j + 1]
+
+            # quadratic parts of penalty
+            for (j, z) in enumerate(rowlist):
+                for k in range(j, len(rowlist)):
+                    # if j and k already exist in the quadratic terms dict,
+                    # add a penalty term into existing value
+                    # else create new key and value in the quadratic term dict
+
+                    if conv_matrix[j + 1][k + 1] != 0:
+                        tup = cast(Union[Tuple[int, int], Tuple[str, str]], (z[0], rowlist[k][0]))
+                        quadratic[tup] = (
+                            quadratic.get(tup, 0.0) + sense * penalty * conv_matrix[j + 1][k + 1]
+                        )
+
+        # Copy quadratic_constraints
+        for quadratic_constraint in problem.quadratic_constraints:
+            self._dst.quadratic_constraints.append(quadratic_constraint)
+
+        if problem.objective.sense == QuadraticObjective.Sense.MINIMIZE:
+            self._dst.minimize(offset, linear, quadratic)
+        else:
+            self._dst.maximize(offset, linear, quadratic)
+
+        # Update the penalty to the one just used
+        self._penalty = penalty  # type: float
+
+        return self._dst
+
+    def _conversion_matrix(self, constraint) -> np.ndarray:
+        """Construct conversion matrix for special constraint.
+
+        Returns:
+            Return conversion matrix which is used to construct
+            penalty term in main function.
+
+        """
+
+        vars_dict = constraint.linear.to_dict(use_name=True)
+        vrs = list(vars_dict.items())
+        rhs = constraint.rhs
+        sense = constraint.sense
+
+        num_vars = len(vrs)
+        combinations = list(itertools.combinations(np.arange(num_vars), 2))
+
+        # conversion matrix
+        conv_matrix = np.zeros((num_vars + 1, num_vars + 1), dtype=int)
+
+        for combination in combinations:
+            index1 = combination[0] + 1
+            index2 = combination[1] + 1
+
+            if rhs in (1, num_vars - 1):
```

Comment:
It is not clear which part corresponds to which inequality pattern. Could you write a comment as same as `_is_special_constraint`, e.g., `# x + y <= 1`, etc.

Reply:
Added a comment which inequality constraint matches.

Reply:
Could you write comments for `if rhs in (1, num_vars - 1)` section?

Reply:
Added comment to the if statement.

Reply:
I would like you to add `x_1 + ... + x_n <= 1` or `x_1 + ... + x_n >= n - 1` in comments.
You can simplify the codes as follows. The current code is a bit tricky.
```
if sense == GE and rhs == num_vars - 1:
  # x_1 + ... + x_n >= n - 1
  ...
elif sense == LE and rhs == 1:
  # x_1 + ... + x_n <= 1
  ...
elif rhs == 0:
  if sense == GE:
    # x >= y
    ...
  elif sense == LE:
    # x <= y
    ...
```
  ...

Reply:
Simplified according the suggestion.